var/spack/repos/builtin/packages/amdlibm/package.py

```diff
@@ -58,6 +58,7 @@ class Amdlibm(SConsPackage):
 
     patch("0001-libm-ose-Scripts-cleanup-pyc-files.patch", when="@2.2")
     patch("0002-libm-ose-prevent-log-v3.c-from-building.patch", when="@2.2")
+    patch("libm-ose-SconsSpack.patch", when="@3.1:4.2")
```

Comment:
Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

Reply:
Also, just for my understanding, why the build system needs to be this tightly coupled to Spack (to the point it must know which env variables are set)?

Reply:
1. What is the effect of the patch?

- Propagates a needed env var back into SCONS which deletes them all by default

2. Why it is needed ?

- It's the only way we see to make the build system work

3. Where it was taken from

- Backport of an internal patch to released versions

4. why the build system needs to be this tightly coupled to Spack (to the point it must know which env variables are set)?

- Per @trws - we don't want it to be, but we have no choice but to repropagate the required env vars.

Reply:
@amd-toolchain-support Thanks for the explanation, but:

> Can you add **a brief comment above the patch directive** explaining:

can you write a comment above the patch directive? It helps when reading packages.